up-l1/zenoh.adoc

```diff
@@ -35,9 +35,93 @@ Zero Overhead Network Protocol.
 
 Zenoh /zeno/ is a pub/sub/query protocol unifying data in motion, data at rest and computations. It elegantly blends traditional pub/sub with geo distributed storage, queries and computations, while retaining a level of time and space efficiency that is well beyond any of the mainstream stacks.
 
-
-for more information, please visit https://projects.eclipse.org/projects/iot.zenoh or https://zenoh.io/
+For more information, please visit https://projects.eclipse.org/projects/iot.zenoh or https://zenoh.io/
 
 == Specifications
 
-_Coming Soon_
\ No newline at end of file
+=== Zenoh Config
+
+While constructing the UPClientZenoh object, the Zenoh config needs to be assigned.
+
+=== UAttribute Mapping
+
+Zenoh supports user attachment.
+We can send additional information with the mechanism, for example, UAttribute.
+This can reduce the unnecessary serialization on payload, which mostly takes time.
+
+User attachment supports key-value pairs, and the following table is the key-value pairs using now.
+
+[cols="1,1"]
+|===
+| key | value
+
+| "uattr"
+| UAttribute object encoded into protobuf
+|===
+
+Besides the user attachment, some features in the UAttribute can be mapped into Zenoh transport settings.
+
+==== Message Type
+
+There are 3 kinds of message types in uProtocol (publish, request, response).
+Different Zenoh operations should use the corresponding message types.
+
+[cols="1,1"]
+|===
+| uProtocol message type | Zenoh operations
+
+| publish | put
+| request | get (query)
+| response | reply (queryable)
+|===
+
+==== Priority Mapping (uProtocol to Zenoh):
+
+[cols="1,1"]
+|===
+| uProtocol priority | Zenoh Priority
+
+| CS0 | BACKGROUND
+| CS1 | DATA_LOW
+| CS2 | DATA
+| CS3 | DATA_HIGH
+| CS4 | INTERACTIVE_LOW
+| CS5 | INTERACTIVE_HIGH
+| CS6 | REAL_TIME
+|===
+
+Note that the priority in Zenoh only works for publisher.
+
+=== URI Mapping
+
+UUri should be able to be serialized it into micro form.
+The mapping from micro-from UUri to Zenoh key is `Vec<u8>` to lower hex string.
+
+Zenoh key for uProtocol MUST start with `up/`.
+If UAuthority is defined in UUri, it should be put in the following part.
+Then the rest of the UUri can be put in the last part.
+
+* UUri without UAthourity: `up/[byte representation of the UUri]`
+* UUri with UAthourity: `up/[byte representation of the UAuthority]/[byte representation of the rest of the UUri]`
+
+Note that if the UUri only contains UAthority with the empty UEntity and UResource,
+it's defined as a special form of Zenoh key `up/[byte representation of UAuthority]/**`.
+This is only used by uStreamer.
+
```

Comment:
I think it'd be good to add a section to the bottom here where we explain
1. what a UUri without UAuthority and with UAuthority look like (written out in psuedocode for .proto maybe, since there will be different implementation languages)
2. what the byte array representation would look like i.e. `[22, 33, 128]`
3. and then what the final Zenoh KeyExpr string looks like, i.e. `up/actual-byte-strings/more-actual-byte-strings`.

For me at least, I think it'll help clarify. WDYT?

Reply:
Done.